js/common/lib/tensor-impl.ts

```diff
@@ -146,8 +146,8 @@ export class Tensor implements TensorInterface {
               // Throw error here because when user try to use number array as data,
               // e.g. new Tensor('float16', [1, 2, 3, 4], dims)), it will actually call
               // Uint16Array.from(arg1) which generates wrong data.
-              throw new TypeError(
-                  'Creating a float16 tensor from number array is not supported. Please use Uint16Array as data.');
+              // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+              data = (typedArrayConstructor as any).from(arg1);
```

Comment:
as explained in the old error message, you cannot do `typedArrayConstructor.from` because it does not work with Uint16Array. you need to check if typedArrayConstructor is Float16Array.